doc/userguide/src/ExtendingRobotFramework/CreatingTestLibraries.rst

```diff
@@ -1667,6 +1667,10 @@ __ `Setting custom name`_
 __ `Embedding arguments into keyword names`_
 __ `Specifying argument types using @keyword decorator`_
 
+.. note:: Storing original keyword name, as ``sourcename`` property
+          when it contains embedded arguments is a new feature in Robot Framework 4.0.
+          It allows for easier matching keywords definition with data from output.xml files.
+
```

Comment:
- I don't think this should be a note because this topic hasn't been discussed otherwise. Could be just a regular paragraph explaining that the original name of the keyword is stored as `sourcename` in the output file starting from RF 4.0. Or at least the note should be worded differently like "To ease finding keyword usages from output files, the original name of the keyword is stored ....".
- This should be explained in the section discussing embedded arguments, not in the general `@keyword` decorator section. This applies also to user keywords so perhaps this should be mentioned in the section discussing using embedded args, not here in the creating libraries section.
- This is documented in the schema so I'm not 100% sure is a separate doc needed here in the UG at all. A short paragraph/note certainly wouldn't hurt.